src/onboarding/map.ts

```diff
@@ -0,0 +1,163 @@
+import { initializeAccount } from 'src/account/actions'
+import {
+  choseToRestoreAccountSelector,
+  recoveringFromStoreWipeSelector,
+} from 'src/account/selectors'
+import {
+  numberVerifiedCentrallySelector,
+  skipVerificationSelector,
+  supportedBiometryTypeSelector,
+} from 'src/app/selectors'
+import { setHasSeenVerificationNux } from 'src/identity/actions'
+import * as NavigationService from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { store } from 'src/redux/store'
+
+interface GetOnboardingMapProps {
+  navigate: typeof NavigationService.navigate | ((screen: Screens) => void)
+  popToScreen: typeof NavigationService.popToScreen | ((screen: Screens) => void)
+  navigateHome: typeof NavigationService.navigateHome | (() => void)
+  navigateClearingStack:
+    | typeof NavigationService.navigateClearingStack
+    | ((screen: Screens) => void)
+  dispatch: (action: any) => void
+}
+
+/**
+ * Traverses through the directed graph of onboarding navigate, navigateClearingStack, and navigateHome calls
+ * and count the number of screens until the given screen and the total number
+ */
+export function getStepValues(screen: Screens) {
+  let stepCounter = 0
+  let totalCounter = 0
+  let reachedScreeen = false
+  let currentScreen: Screens = Screens.NameAndPicture
```

Comment:
would be nice if `Screens.NameAndPicture` was a constant somewhere named FIRST_ONBOARDING_SCREEN or something